src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -33,8 +33,14 @@ public class ContainerOperationProvider : RunnerService, IContainerOperationProv
         public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
-            _dockerManager = HostContext.GetService<IDockerCommandManager>();
-            _containerHookManager = HostContext.GetService<IContainerHookManager>();
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.Hooks.ContainerHooksPath)))
```

Comment:
Should we do `same` in the `ContainerActionHandler`?

https://github.com/actions/runner/blob/main/src/Runner.Worker/Handlers/ContainerActionHandler.cs#L41

Reply:
Great catch! Thanks!